File: app/schemas/image.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ImageResponse(BaseModel):

    id: int

    product_id: int

    image_url: str

    filename: str

    mime_type: str

    file_size: int

    width: int

    height: int

    display_order: int

    created_at: datetime

    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )

File: frontend/src/types/image.ts
export interface ProductImage {
  id: number;
  product_id: number;
  image_url: string;
  filename: string;
  mime_type: string;
  file_size: number;
  width: number;
  height: number;
  display_order: number;
  created_at: string;
  updated_at: string;
}

export interface ImageDisplayOrderUpdate {
  display_order: number;
}
